README.md
# Mods: Factorio: Wood Gasification

<img align="right" width="100" height="100" src="https://github.com/cappie/mods-factorio-Wood_Gasification/blob/master/thumbnail.png">
This is the repository for the Wood Gasification mod for Factorio.

Too much wood? No oil on your map? Want petroleum gas, solid fuel or crude oil? Then this mod's for you!

Wood gasification is the process of turning wood into it's carbohydrate constituents and releasing its chemical potential in the form of several oil products: petroleum gas, diluted bitumen, which can be converted to crude oil.

Wood gasification comes in two flavors: Wood gas processing uses a chemical lab to turn 10 units of wood into 20 units of petroleum gas and 8 units of diluted bitumen. 32 units of diluted bitumen can be turned into 1 unit of solid fuel. With some further research, you can turn 32 diluted bitumen and 200 water into 40 crude oil. At a higher research level, you'll have advanced wood gasification processes that uses a refinery to turn 20 wood and 200 water into 15 units of heavy oil, 15 units of light oil and 30 petroleum gas. This mod is meant to kick-start your chemical plants (or when you have a shortage of crude oil OR a surplus of wood on your map)

This mod has been translated to these languages:

- English (cappie)
- Dutch (cappie)
- German (Reysuke)
- Russian (Va7ya)
- French (kryshnar & meah41)
- Polish (Mgis)
- Spanish (Deepl.com)
- Italian (Deepl.com)
- Portuguese (Deepl.com)

## Continuation of unmaintained mod

This mod is an update for the 0.14 mod 'Wood Gas' which was discontinued; It's my attempt to upgrade 'Wood Gas' (0.14 only) to 0.15+ and continue the development of the mod. The original author ('mooklepticon') of 'Wood Gas' has given me his blessing to continue this mod.

## Translators wanted

I would love to be able to support even more languages like Spanish, Portugese, Italian, Japanese and maybe some slavic languages and perhaps even Hungarian or Finnish?. If you would like to help, please take a look at the repository, or if you don't know how to do that Github thing, just contact me via <https://mods.factorio.com/> and I'll add your translated strings from the locale-file to the mods' repository.

## TLDR

This mod processes wood into oil products.

- Early game: Wood in Chemical plant creates Petroleum Gas & Diluted Bitumen. Diluted Bitumen & Water in Chemical plant produces either Crude Oil or Solid Fuel
- Later game: Wood in Refinery creates Heavy Oil, Light Oil & Petroleum Gas directly.


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 2.6.0
Date: 2021-03-08
  Changes:
    - New tar, recipes and tech icons by snouz
    - New mod icon
    - icon mipmaps
---------------------------------------------------------------------------------------------------
Version: 2.5.0
Date: 2020-03-18
  Changes:
    - Added Spanish, Italian and Portuguese translations using Deepl.com
    - Changed some Dutch translations
  Bugfixes:
    - Fixed my broken changelog (I REALLY need automated tests and a changelog linter for these factorio mods)
---------------------------------------------------------------------------------------------------
Version: 2.4.0
Date: 2020-03-14
  Changes:
    - Added Polish and French translations
---------------------------------------------------------------------------------------------------
Version: 2.3.0
Date: 2020-02-06
  Rebalancing:
    - Wood gasification costs 100 automation and logistic science (reasonably simple to research)
    - Wood gasification of bitumen to oil costs 400 automation and logistic science (it's simple science, but hard to perfect)
    - Advanced wood gasification costs 150 automation, logistic and chemical science (wood-to-oil easy-mode will take some effort)
    - Wood gasification uses 10 wood to produce 20 petroleum gas and 8 bitumen (80% efficient)
    - Wood gasification enables you to convert 32 bitumen into 1 solid fuel (75% efficient)
    - Advanced bitumen processing uses 32 bitumen and 200 water to produce 40 crude oil (90% efficient (crude oil takes another efficiency hit during refining))
    - Advanced wood gasification uses 20 wood and 200 water to produce 15 heavy oil, 15 light oil and 30 petroleum gas (90% efficient)
  Changes:
    - Changed the text string of Tar into Diluted Bitumen to be somewhat more scientifically correct, internally it's all still 'tar' for brevity.
    - Added bitumen processing to crude oil using bitumen upgrading so you can turn wood into crude oil
    - Added wood-gasification-to-oil as a new researchable technology after advanced wood gasification
    - Updated the balancing.xls to show the calculations behind the wood-gasification-to-oil recipe
    - Updated the source.psd after creating the graphics for the new wood-gasification-to-oil recipe and technology
  Bugfixes:
    - Fixed wood (and water) showing up as raw product for other oil-related products (this should be crude oil, not wood)
    - Fixed the graphics by removing the coal that was still in there from older versions
  Notes:
    - Just FYI: this version of the mod still works with 0.17.x
  Broke:
    - The Russian and German translations probably need work since I used Google translate for the new tech.
    - I tried not to break too much, but your wood processing setups probably need some love and attention.
---------------------------------------------------------------------------------------------------
Version: 2.2.0
Date: 2020-02-06
  Changes:
    - Made the mod work with 0.18.x (sorry for being late to the party)
---------------------------------------------------------------------------------------------------
Version: 2.1.3
Date: 2020-02-06
  Rebalancing:
    - Did the same rebalancing for 0.17.x as for 0.18.x in 2.3.0
---------------------------------------------------------------------------------------------------
Version: 2.1.2
Date: 2019-11-18
  Changes:
    - Added 'Holz' to the german translations
    - Updated the GIT repository with the actual mod
  Bugfixes:
    - Fixed minor readme inconsistencies while updating the GIT repo
---------------------------------------------------------------------------------------------------
Version: 2.1.1
Date: 2019-09-10
  Changes:
    - Fixed the english language strings (petroleum was spelled wrong, thanks for noticing, tyrannus1)
    - Added german translations (thank you for translating, tyrannus1)
    - Added russian translations (thank you for translating, Va7ya)
---------------------------------------------------------------------------------------------------
Version: 2.1.0
Date: 2019-07-19
  Changes:
    - Added advanced wood gasification using a refinery (28 wood & 50 water generates 35 heavy oil, 15 light and 20 petroleum gas
    - Added some nicer images for the technologies
    - Added calculations to the balancing.xls in the dev folder
    - Removed the dev folder from the mod .zip archive (it's still available on GitHub, but the 250kb doesn't belong in the mod itself
  Broke:
    - Broke the german translations; feel free to retranslate them (Reysuke?)
---------------------------------------------------------------------------------------------------
Version: 2.0.4
Date: 2019-03-21
  Changes:
    - Added Dutch and German translations. (Thanks to Reysuke for the German translations)
  Bugfixes:
    - Fixed the English language strings to better facilitate translations.
    - Fixed fluid tar properties to better reflect the actual properties of tar (not sure if this helps the game or not)
---------------------------------------------------------------------------------------------------
Version: 2.0.3
Date: 2019-03-10
  Changes:
    - Got in a fight with mods.factorio.com, but it should be ok now.
---------------------------------------------------------------------------------------------------
Version: 2.0.2
Date: 2019-03-10
  Changes:
    - Removed all the old 0.16 references; this mod is now 0.17+ only!
---------------------------------------------------------------------------------------------------
Version: 2.0.1
Date: 2019-03-10
  Bugfixes:
    - Fixed technology description to reflect the new balancing.
    - Fixed changelog.txt error which caused it not being able to parse (thank for catching the mistake, Pi-C!)

prototypes/technology.lua
data:extend({
  {
    type = "technology",
    name = "wood-gas-processing",
    icon = "__Wood_Gasification__/graphics/technology/wood-gas-processing.png",
    icon_size = 256, icon_mipmaps = 4,
    prerequisites = {"oil-processing"},
    effects =
    {
      {
        type = "unlock-recipe",
        recipe = "wood-gasification",
      },
	  {
        type = "unlock-recipe",
        recipe = "solid-fuel-from-tar"
      }, 
    },
    unit =
    {
      count = 100,
      ingredients = {
        {"automation-science-pack", 1},
        {"logistic-science-pack", 1}},
      time = 30
    },
    order = "d-b"
  },
  {
    type = "technology",
    name = "wood-gas-processing-to-crude-oil",
    icon = "__Wood_Gasification__/graphics/technology/wood-gas-processing-to-crude-oil.png",
    icon_size = 256, icon_mipmaps = 4,
    prerequisites = {"wood-gas-processing"},
    effects =
    {
      {
        type = "unlock-recipe",
        recipe = "crude-oil-from-tar",
      },
    },
    unit =
    {
      count = 400,
      ingredients = {
        {"automation-science-pack", 1},
        {"logistic-science-pack", 1}
      },
      time = 30
    },
    order = "d-b"
  },
  {
    type = "technology",
    name = "advanced-wood-gas-processing",
    icon = "__Wood_Gasification__/graphics/technology/advanced-wood-gas-processing.png",
    icon_size = 256, icon_mipmaps = 4,
    prerequisites = {"wood-gas-processing"},
    effects =
    {
      {
        type = "unlock-recipe",
        recipe = "advanced-wood-gasification",
      }
    },
    unit =
    {
      count = 150,
      ingredients = {
        {"automation-science-pack", 1},
        {"logistic-science-pack", 1},
        {"chemical-science-pack", 1}
      },
      time = 30
    },
    order = "d-b"
  },
})

prototypes/recipe.lua
data:extend({
    {
        type = "recipe",
        name = "wood-gasification",
        category = "chemistry",
        enabled = false,
        energy_required = 5,
        ingredients = {
            {type="item", name="wood", amount=10},
        },
        results = {
            {type="fluid", name="petroleum-gas", amount=20},
            {type="fluid", name="tar", amount=8},
        },
        icon = "__Wood_Gasification__/graphics/wood-gasification.png",
        icon_size = 64,
        subgroup = "fluid-recipes",
        order = "a[fluid-chemistry]-w[wood-gas-processing]",
        crafting_machine_tint =
        {
          primary = {r = 0.698, g = 0.698, b = 0.698, a = 0.000}, -- #7f7f7f00
          secondary = {r = 0.400, g = 0.400, b = 0.400, a = 0.000}, -- #66666600
          tertiary = {r = 0.305, g = 0.305, b = 0.305, a = 0.000}, -- #4d4d4d00
        },
        always_show_products = true,
        always_show_made_in = true,
        allow_decomposition = false,
		order = "a-w"
    },
    {
        type = "recipe",
        name = "solid-fuel-from-tar",
        category = "chemistry",
        enabled = false,
        energy_required = 5,
        ingredients = {
            {type="fluid", name="tar", amount=32},
        },
        results = {
            {type="item", name="solid-fuel", amount=1},
        },
        icon = "__Wood_Gasification__/graphics/solid-fuel-from-tar.png",
        icon_size = 64,
        subgroup = "fluid-recipes",
        order = "a[fluid-chemistry]-w[solid-fuel-from-tar]",
        crafting_machine_tint =
        {
            primary = {r = 0.000, g = 0.000, b = 0.000, a = 0.000}, -- #00000000
            secondary = {r = 0.000, g = 0.000, b = 0.000, a = 0.000}, -- #00000000
            tertiary = {r = 0.000, g = 0.000, b = 0.000, a = 0.000}, -- #00000000
        },
        always_show_products = true,
        always_show_made_in = true,
		allow_decomposition = false,
		order = "a-w"
    },
    {
        type = "recipe",
        name = "crude-oil-from-tar",
        category = "chemistry",
        enabled = false,
        energy_required = 5,
        ingredients = {
            {type="fluid", name="tar", amount=32},
            {type="fluid", name="water", amount=200},
        },
        results = {
            {type="fluid", name="crude-oil", amount=40},
        },
        icon = "__Wood_Gasification__/graphics/crude-oil-from-tar.png",
        icon_size = 64,
        subgroup = "fluid-recipes",
        order = "a[fluid-chemistry]-w[crude-oil-from-tar]",
        crafting_machine_tint =
        {
            primary = {r = 0.290, g = 0.027, b = 0.000, a = 0.000}, -- #49060000
            secondary = {r = 0.722, g = 0.465, b = 0.190, a = 0.000}, -- #b8763000
            tertiary = {r = 0.870, g = 0.365, b = 0.000, a = 0.000}, -- #dd5d0000
        },
        always_show_products = true,
        always_show_made_in = true,
		allow_decomposition = false,
		order = "a-w"
    },
    {
        type = "recipe",
        name = "advanced-wood-gasification",
        category = "oil-processing",
        enabled = false,
        energy_required = 10,
        ingredients =
        {
          {type="item", name="wood", amount=20},
          {type="fluid", name="water", amount=200}
        },
        results=
        {
          {type="fluid", name="heavy-oil", amount=15},
          {type="fluid", name="light-oil", amount=15},
          {type="fluid", name="petroleum-gas", amount=30}
        },
        icon = "__Wood_Gasification__/graphics/advanced-wood-gasification.png",
        icon_size = 64,
        subgroup = "fluid-recipes",
        order = "a[oil-processing]-w[advanced-wood-gasification]",
        always_show_products = true,
        always_show_made_in = true,
        allow_decomposition = false,
		order = "a-w"
    },
})